openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -82,6 +83,12 @@ def content_type_gating_enabled(self):
             course_key=self.course_key,
         )
 
+    @property
+    def can_show_upgrade_sock(self):
+        enrollment = CourseEnrollment.get_enrollment(self.effective_user, self.course_key)
```

Comment:
And just to note, we can _not_ use the existing `self.enrollment` property, because it only provides a subset of the data needed here (the whole enrollment object).

Reply:
Could we save more data in self.enrollment and pare it down when it's time to send it over the wire to avoid the extra call?

Reply:
Probably doable, but would need some refactoring. It actually calls `CourseEnrollment.enrollment_mode_for_user` and not `get_enrollment`.